server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/spec/pipeline_config_create_widget_spec.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PipelineConfig} from "models/pipeline_configs/pipeline_config";
+import _ from "lodash";
+import {PipelineConfigCreateWidget} from "views/pages/pipeline_configs/pipeline_config_create_widget";
+import {TestHelper} from "views/pages/spec/test_helper";
+import m from "mithril";
+
+describe("PipelineCreateWidgetSpec", () => {
+  let helper       = new TestHelper();
+  const noop       = _.noop;
+  const operations = {
+    onDelete: noop,
+    onAdd: noop
+  };
+
+  it("should render pipeline name field", function () {
+    helper.mount(() => <PipelineConfigCreateWidget materialOperations={operations}
+                                                   pipelineConfig={new PipelineConfig("", [], [])}/>);
+
+    debugger;
+
+    expect(helper.findByDataTestId("form-field-label-pipeline-name")).toContainText("Pipeline name");
+    expect(helper.findByDataTestId("pipeline-name-input")).toHaveLength(1);
+    expect(helper.findByDataTestId("pipeline-name-input")).toHaveProp("required");
+    expect(helper.findByDataTestId("pipeline-details-container"))
+      .toContainText("No spaces. Only letters, numbers, hyphens, underscores and period. Max 255 chars");
```

Comment:
extract `"No spaces. Only letters, numbers, hyphens, underscores and period. Max 255 chars" ` into a variable?